[SFT-1608]: Form Monitor

gustavs-gutmanis opened the pull request:
### Related Ticket Number
[SFT-1608](https://solspace.atlassian.net/browse/SFT-1608)

### Description
Integrating Form Monitor into Freeform


[SFT-1608]: https://solspace.atlassian.net/browse/SFT-1608?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

gustavs-gutmanis committed:
feat(SFT-1608): adding initial Form Monitor files

gustavs-gutmanis committed:
feat(SFT-1608): adding initial Form Monitor files

gustavs-gutmanis committed:
feat(SFT-1608): add form monitor integration

gustavs-gutmanis committed:
feat(SFT-1608): add form monitor integration

gustavs-gutmanis committed:
chore: updating artifacts

gustavs-gutmanis committed:
chore: updating artifacts

gustavs-gutmanis committed:
feat(SFT-1608): adding form manifest

gustavs-gutmanis committed:
feat(SFT-1608): adding form manifest

gustavs-gutmanis committed:
feat(SFT-1608): sending URL with manifest

gustavs-gutmanis committed:
feat(SFT-1608): sending URL with manifest

gustavs-gutmanis committed:
feat(SFT-1608): adding captcha bypass

gustavs-gutmanis committed:
feat(SFT-1608): adding captcha bypass

gustavs-gutmanis committed:
feat(SFT-1608): add hidden submission support for form monitor checking

gustavs-gutmanis committed:
feat(SFT-1608): add hidden submission support for form monitor checking

gustavs-gutmanis committed:
Merge remote-tracking branch 'origin/feat/SFT-1608-form-monitor' into feat/SFT-1608-form-monitor

gustavs-gutmanis committed:
feat(SFT-1608): disabling integrations for form monitor requests

gustavs-gutmanis committed:
feat(SFT-1611): override all emails when submitting through FM

gustavs-gutmanis committed:
feat(SFT-1613): build UI for Form Monitor

seandelaney commented on packages/client/src/app/pages/form-monitor/form-monitor.tsx:
```diff
@@ -0,0 +1,62 @@
+  useSidebarSelect(5);
+
+  const { data: forms, isFetching: isFetchingForms } = useQueryFormsWithStats();
+  const { data: formIds, isFetching: isFetchingFormids } = useFMForms();
```
`isFetchingFormids` --> `isFetchingFormIds`

seandelaney replied:
Also not liking the use of `FM` instead of `FormMonitor` throughout the API

gustavs-gutmanis replied:
Good pointers, I'll change them

seandelaney commented on packages/plugin/src/Integrations/Other/FormMonitor/FormMonitor.php:
```diff
@@ -0,0 +1,108 @@
+    public function getApiRootUrl(): string
+    {
+        return 'https://api.leftip.com/v1';
+        // return 'https://api.formmonitor.com/v1';
```
What is `leftip.com` ?

Add FIXME so we don't forget

gustavs-gutmanis replied:
Test domain while the development is ongoing.

seandelaney reviewed (dismissed):
So far functionality is all working. No issues adding new Form Monitor integration to existing forms.

gustavs-gutmanis committed:
Merge branch 'v5' into feat/SFT-1608-form-monitor

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	packages/plugin/src/Resources/js/client/client.js

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

gustavs-gutmanis committed:
feat(SFT-1608): add submission ack call

gustavs-gutmanis committed:
fix(SFT-1608): broken tests page

gustavs-gutmanis committed:
fix(SFT-1608): sidebar navigation items

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

kjmartens committed:
Merge branch 'feat/v5.9' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	packages/plugin/src/Models/Pro/ExportProfileModel.php
#	packages/plugin/src/Resources/js/client/client.js

kjmartens committed:
Bump version

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	composer.json
#	packages/plugin/src/Resources/js/client/client.js

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	packages/plugin/src/Resources/js/client/client.js

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	packages/plugin/src/Resources/js/client/client.js

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	packages/plugin/src/Resources/js/client/client.js

thejahid committed:
feat(SFT-1608): Implement Form Monitor stats and UI improvements

kjmartens committed:
Updating version number

thejahid committed:
feat(SFT-1608): Enhance Form Monitor with pagination, error handling, and detailed test views

thejahid committed:
feat(SFT-1608): Add Form Monitor integration to form builder and list views

- Implemented Form Monitor sidebar and results view in form builder
- Added Form Monitor stats to form list card view
- Created new types and queries for Form Monitor data
- Integrated Form Monitor service in plugin backend
- Removed previous Form Monitor components

thejahid committed:
fix(SFT-1811): Update loader background styling

thejahid committed:
fix(SFT-1608): Improve Form Monitor integration removal logic

thejahid committed:
feat(SFT-1789): Exclude hidden submissions from charts and count queries

thejahid committed:
feat(SFT-1820): Add Form Monitor column to forms list view

thejahid committed:
feat(SFT-1786): Improve Form Monitor date formatting across client and plugin

thejahid committed:
feat(SFT-1820): Enhance Form Monitor stats visualization in forms list view

- Added a new stats chart component for Form Monitor in list view
- Implemented a gradient line indicator showing success, failed, and pending percentages
- Updated styling for monitor status and stats display
- Removed unused list item styles file

thejahid committed:
feat(SFT-1812): Add status icons and display for most recent form tests

thejahid committed:
feat(SFT-1608): Refactor status display styles and logic for form monitor

kjmartens committed:
feat(SFT-1608): Adjustments to colors, switching “Pending” to “Processing”

thejahid committed:
feat(SFT-1824): Display previous test status in form monitor for better context

thejahid committed:
feat(SFT-1608): build client

thejahid committed:
feat(SFT-1789): Exclude hidden submissions from charts queries

gustavs-gutmanis committed:
feat(SFT-1831): moving FormMonitor to singleton

thejahid committed:
feat(SFT-1826): Enhance Form Monitor UI with comprehensive test results and configuration panel

- Added new styled components for status indicators and dots
- Implemented detailed test results view with improved visualization
- Created a configuration panel showing integration and service status
- Updated sidebar to display more comprehensive test information
- Refactored test results rendering with better status representation
- Added support for displaying monitored URL and integration details

thejahid committed:
feat(SFT-1841): Optimize UI layout and styling for test results and stats

thejahid committed:
fix(SFT: 1762): Normalize SQL condition case for request ID check

thejahid committed:
feat(SFT-1848): Improve Form Card Layout and Responsiveness

thejahid committed:
fix(SFT-1762): Improve SQL query table name reference for request ID check

thejahid committed:
fix(SFT-1856): Remove previous test references

kjmartens committed:
fix: bundle loader crashing if a cached class is found in the class map (#1823)

thejahid committed:
Merge branch 'feat/SFT-1608-form-monitor' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

thejahid committed:
fix(FormMonitorCleanupJob): Add early return if no submissions exist to prevent unnecessary processing

thejahid committed:
feat(SFT-1886): Implement service reactivation functionality in Form Monitor

- Added a button to reactivate the service when inactive
- Introduced status messages for loading, success, and error states during reactivation
- Updated configuration panel to include form ID and refetch data after reactivation
- Created a new mutation hook for enabling form monitoring via API
- Enhanced styling for status indicators and buttons

thejahid committed:
feat(SFT-1898): Add Next Scheduled Test Panel to Form Monitor

thejahid committed:
Merge branch 'v5' into feat/SFT-1608-form-monitor

thejahid committed:
fix(SFT-1887): adjust TestStatusIcon size and update failed icon SVG dimensions

thejahid committed:
fix(SFT-1902): include form name in payload for FormMonitor integration

thejahid committed:
feat(SFT-1893): add delete test functionality in FormMonitor integration

thejahid committed:
feat(SFT-1893): build client

kjmartens committed:
chore(SFT-1608): adjusting dashboard table heading

kjmartens committed:
chore(SFT-1608): changing icon

thejahid committed:
feat(SFT-1922): Add a "Beta" flag to Monitoring tab

thejahid committed:
Merge branch 'feat/SFT-1608-form-monitor' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

thejahid committed:
feat(SFT-1922): build client

kjmartens committed:
Adjustments to wording

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	packages/plugin/src/Resources/js/client/client.js

thejahid committed:
feat(SFT-1893): add clear all tests functionality and related UI components

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	packages/plugin/src/Resources/js/client/client.js

thejahid committed:
refactor(SFT-1940): enhance delete confirmation logic and UI for bulk deletion; update styles for ClearAllButton

thejahid committed:
feat(SFT-1920): updates to monitoring tab

thejahid committed:
feat(SFT-1925): update form monitor results and sidebar logic for improved status handling and messaging

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	packages/plugin/src/Resources/js/client/client.js

thejahid committed:
fix(SFT-1955): reverse test display order in DailyTestColumn for improved visibility

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	packages/plugin/src/Resources/js/client/client.js
#	packages/plugin/src/Resources/js/client/vendor.js

thejahid committed:
fix(SFT-1964): remove gap from CardWrapper styles and update monitor status text to 'Pending'

kjmartens committed:
feat(SFT-1970): adjust height of FM chart

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	composer.json
#	packages/plugin/src/Resources/js/client/client.js

kjmartens committed:
fix: diag attempt to read property "name" on null for FM

thejahid committed:
refactor(FormMonitor): remove unused deleteTest action and clean up API root URL formatting

thejahid committed:
feat(SFT-1973): enhance status indicators with new styles and animations; add new icons for success and error states

thejahid committed:
feat(SFT-1974): update monitoring stats component with new styles and functionality; integrate last test status display and improve layout

thejahid committed:
feat(SFT-1979): add 'uid' property to transformed field properties for enhanced tracking

thejahid committed:
feat(SFT-1992): adjust screenshot button visibility logic in results component

thejahid committed:
fix(SFT-1996): Uptime pending one line

gustavs-gutmanis committed:
fix(SFT-1982): fixing merge issue with export profiles

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	composer.json
#	packages/plugin/src/Resources/css/cp/integrations/singleton.css
#	packages/plugin/src/Resources/js/client/client.js

gustavs-gutmanis committed:
fix(SFT-1983): display integration errors in builder when saving

kjmartens committed:
Merge branch 'feat/SFT-1608-form-monitor' of https://github.com/solspace/craft-freeform into fix/SFT-1983-integration-errors

# Conflicts:
#	packages/plugin/src/Resources/js/client/client.js

thejahid committed:
feat(SFT-1983): Show integration error to the builder & check max forms at formmonitor

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	composer.json
#	packages/plugin/src/Resources/js/client/client.js

thejahid committed:
feat(SFT-1997): Prepopulate the `Error Notification Email` setting - Email

thejahid committed:
feat(SFT-1997): add ValueGenerator import to FormMonitor for enhanced functionality

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

thejahid committed:
fix: update scan waiting message to reflect shorter expected duration

kjmartens committed:
chore: correct typo

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	composer.json

thejahid committed:
feat(SFT-1995): update email recipient in EmailOverrides to use inbound address

thejahid committed:
Merge branch 'feat/SFT-1608-form-monitor' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	composer.json

kjmartens committed:
Initial Readme for FM

thejahid committed:
feat(SFT-1995): add notification support to form monitor results, including display of enabled notifications and types

thejahid committed:
Merge branch 'feat/SFT-1608-form-monitor' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

kjmartens committed:
chore: updated Readme for FM

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

kjmartens committed:
chore: linting

thejahid committed:
feat(SFT-1994): Add ability for customers to "Disable" testing AND "Delete" an entire form's connection/data to FM tab in builder

thejahid committed:
feat(SFT-2059): "Disable & Delete Monitoring Data" does not remove the Form record in FM

thejahid committed:
feat(SFT-2052): enhance form monitor results with detailed notification statistics and status indicators

thejahid committed:
fix(SFT-2070): update order of 'Test Email Notifications on Live Environment only' setting and clean up code structure

thejahid committed:
fix(SFT-2071): add PendingStatus component and update results display for ongoing tests

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	composer.json

kjmartens committed:
fix: adjustments to Form Submit status sizing

gustavs-gutmanis committed:
feat(SFT-1608): prevent FM emails from being sent, when test emails is turned off

gustavs-gutmanis committed:
feat(SFT-1608): add form monitor skip header to conditional notifications too

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	packages/plugin/src/Resources/js/client/client.js

thejahid committed:
fix(SFT-2081): update status indicator logic to handle cases where notifications are greater than or equal to total notifications

thejahid committed:
fix(SFT-2068): enhance response display with tooltip and style adjustments

thejahid committed:
fix(SFT-2068): client building for totalResponse

thejahid committed:
feat(SFT-2082): enhance form monitor UI with new response display and action menu

thejahid committed:
fix(SFT-2083): Enhance error handling in FormMonitorService

thejahid committed:
fix(SFT-2088): Refactor menu items in ConfigurationPanel for better readability

thejahid committed:
fix(SFT-2088): disable & delete monitoring data client

thejahid committed:
fix(SFT-2083): Adjust messaging in Freeform when FM server connection fails

thejahid committed:
feat(SFT-2074): implement lazy loading states for form monitor stats in UI components

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

# Conflicts:
#	composer.json
#	packages/plugin/src/Resources/js/client/client.js

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

kjmartens committed:
fix(#1992): Postgres error when creating new PDF template

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

kjmartens committed:
chore: updated Craft Entry and User integration icons

thejahid committed:
fix(SFT-2104): 🚨 FM appears to break email notifications for NORMAL use of forms

thejahid committed:
Merge branch 'feat/SFT-1608-form-monitor' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

thejahid committed:
fix(SFT-2100): update SVG icons and adjust styles for StatusIcon component

thejahid committed:
feat(SFT-2048): add disable and delete actions with corresponding UI components

kjmartens committed:
feat(SFT-2048): adjust button colors

seandelaney committed:
fix(SFT-2108): added logging to Digest, fixed race condition issue (#2003)

gustavs-gutmanis committed:
fix(SFT-2109): resolved a potential security issue with submission titles (#2002)

kjmartens committed:
v5.10.16

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

thejahid committed:
feat(SFT-1998): Have the "Monitoring" tab magically appear

gustavs-gutmanis committed:
fix: issue with checking lock

kjmartens committed:
v5.10.16.1

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

thejahid committed:
feat(SFT-2116): collect the Craft System Name

kjmartens committed:
chore: cleaning up settings

thejahid committed:
fix(SFT-2124): Any result of 1000+ triggers an error for the Calculation field (#2008)

thejahid committed:
feat(SFT-2121): add time zone information to authorization listener

kjmartens committed:
fix: Site Name should be global property

kjmartens committed:
chore: updating integration README styles

kjmartens committed:
chore: fix link to guide

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

thejahid committed:
fix(SFT-2133): Using .env Variables in Settings does not register in FM

kjmartens committed:
fix: updated the `phpoffice/phpspreadsheet` dependency to include `^4.0`.

thejahid committed:
fix(SFT-2139): update button colors in singleton styles for consistency

seandelaney committed:
fix(SFT-2112): removed sub query causing too many table joins (#2012)

kjmartens committed:
v5.10.17

kjmartens committed:
Merge branch 'v5' of https://github.com/solspace/craft-freeform into feat/SFT-1608-form-monitor

thejahid committed:
feat(SFT-2144): add Craft and Freeform version headers to AuthorizationListener

thejahid committed:
feat(SFT-2148): add DurationBadge component to display submission duration in results
